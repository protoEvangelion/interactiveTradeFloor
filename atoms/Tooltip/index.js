import PropTypes from 'prop-types'
import React from 'react'
import ReactTooltip from 'react-tooltip'

const Tooltip = ({ _id, co, owner, status, description }) => {
  const renderText = () => {
    return status === 'open' || status === 'n/a'
      ? (
        <p key={`span_${_id}`} >
          Status: {status}
        </p>
      )
      : (
        <table key={`span_${_id}`}>
          <tbody>
            <tr>
              <td><strong>Co:</strong></td>
              <td style={{ paddingLeft: '10px' }}>{co}</td>
            </tr>
            <tr>
              <td><strong>Owner:</strong></td>
              <td style={{ paddingLeft: '10px' }}>{owner}</td>
            </tr>
            <tr>
              <td><strong>Status:</strong></td>
              <td style={{ paddingLeft: '10px' }}>{status}</td>
            </tr>
            <tr>
              <td><strong>Info:</strong></td>
              <td style={{ paddingLeft: '10px' }}>{description}</td>
            </tr>
          </tbody>
        </table>
      )
  }
  return (
    <ReactTooltip
      id={`tool_${_id}`}
      type="info"
      effect="solid"
      style={{ opacity: 1 }}
    >
      {renderText()}
    </ReactTooltip>
  )
}

Tooltip.propTypes = {
  _id: PropTypes.string,
  co: PropTypes.string,
  owner: PropTypes.string,
  status: PropTypes.string,
  description: PropTypes.string,
}

export default Tooltip
