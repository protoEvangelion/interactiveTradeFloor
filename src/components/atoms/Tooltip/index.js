import PropTypes from 'prop-types'
import React from 'react'
import ReactTooltip from 'react-tooltip'

const Tooltip = ({ _id, co, owner, status, description }) => {
	const renderText = () => {
		return status === 'open' || status === 'n/a' ? (
			<p key={`span_${_id}`}>Status: {status}</p>
		) : (
			<table key={`span_${_id}`}>
				<tbody>
					<tr>
						<td>
							<strong>Co:</strong>
						</td>
						<td>{co}</td>
					</tr>
					<tr>
						<td>
							<strong>Owner:</strong>
						</td>
						<td>{owner}</td>
					</tr>
					<tr>
						<td>
							<strong>Status:</strong>
						</td>
						<td>{status}</td>
					</tr>
					<tr>
						<td>
							<strong>Info:</strong>
						</td>
						<td>{description}</td>
					</tr>
				</tbody>
			</table>
		)
	}
	return (
		<ReactTooltip
			id={`tool_${_id}`}
			type="info"
			effect="solid"
			style={{ opacity: 1, padding: '8px' }}
		>
			{renderText()}
		</ReactTooltip>
	)
}

Tooltip.propTypes = {
	_id: PropTypes.string,
	co: PropTypes.string,
	owner: PropTypes.string,
	status: PropTypes.string,
	description: PropTypes.string,
}

export default Tooltip
