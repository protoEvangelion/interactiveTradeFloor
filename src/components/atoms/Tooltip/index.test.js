import React from 'react'
import renderer from 'react-test-renderer'
import Tooltip from '.'

const props = {
  _id: '2sdf5d2sdf',
  co: 'AOA',
  owner: 'Ryan',
  status: 'Holding',
  description: 'Stuff',
}

describe('Tooltip snapshot', () => {
  it('renders tooltip', () => {
    const tree = renderer.create(
      <Tooltip {...props} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
