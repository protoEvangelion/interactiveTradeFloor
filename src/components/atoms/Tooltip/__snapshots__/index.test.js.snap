// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> snapshot renders tooltip with holding status 1`] = `
<div
  className="__react_component_tooltip place-top type-dark "
  data-id="tooltip"
>
  
</div>
`;

exports[`<Tooltip /> snapshot renders tooltip with open status 1`] = `
<div
  className="__react_component_tooltip place-top type-dark "
  data-id="tooltip"
>
  
</div>
`;
